cpp/src/arrow/compute/exec/options.h

```diff
@@ -229,6 +229,20 @@ class ARROW_EXPORT SinkNodeConsumer {
   virtual Future<> Finish() = 0;
 };
 
+class ARROW_EXPORT NullSinkNodeConsumer : public SinkNodeConsumer {
```

Comment:
See use in `cpp/src/arrow/engine/substrait/util.cc` for deserializing (and later executing) Substrait plans without a consumer (e.g., just writers). I intend to use this utility from PyArrow code in an upcoming PR.

AFAIU, a Substrait plan must have some consumer for proper execution. Up to now, only a `ConsumingSinkNodeOption` was provided. A `NullSinkNodeConsumer` is useful with the writing capability added here.

Reply:
True, but the method on line 132 is, itself, never used that I can tell.  Also, the docs for that method mention plans with just writes which seems incorrect since those plans would use `WriteNodeFactory`.  If we're going to add new public methods then it should be clear the situation in which the are needed.

Reply:
I moved `NullSinkNodeConsumer` into `serde_test.cc`, which hopefully is acceptable.